File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
import sys
input = sys.stdin.readline
def solution(arr):
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.

    result = [arr[0]]
    
    for i in range(1, len(arr)):
        if result[-1] != arr[i]:
            result.append(arr[i])
    
    return result;

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr) {
    if (arr.length === 0) return [];  

    const result = [];  
    let lastNum = null; 

    for (let num of arr) {
        if (num !== lastNum) { 
            result.push(num);  
            lastNum = num;    
        }
    }

    return result;
}
